301-webapp-storage-keyvault-msi/azuredeploy.json

```diff
@@ -0,0 +1,232 @@
+            "name": "[parameters('webAppName')]",
+            "type": "Microsoft.Web/sites",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2018-02-01",
+            "identity": {
+                "type": "SystemAssigned"
+            },            
+            "properties": {
+                "name": "[parameters('webAppName')]",
+                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('hostingPlanName'))]"
+            },
+            "tags": {
+                "[concat('hidden-related:', resourceGroup().id, '/providers/Microsoft.Web/serverfarms/', parameters('hostingPlanName'))]": "empty",
+                "displayName": "Website"
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.Web/serverFarms/', parameters('hostingPlanName'))]",
+                "[resourceId('Microsoft.Storage/storageAccounts',parameters('storageAccountName'))]",
+                "[resourceId('Microsoft.KeyVault/vaults/',parameters('keyVaultName'))]"
+            ],
+            "resources": [
+                {
+                    "name": "appsettings",
+                    "type": "config",
+                    "apiVersion": "2018-02-01",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/sites', parameters('webAppName'))]"
+                    ],
+                    "tags": {
+                        "displayName": "appSettings"
+                    },
+                    "properties": {
+                        "StorageAccountName":"[parameters('storageAccountName')]",
+                        "StorageAccountUrl": "[reference(concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2018-07-01').primaryEndpoints.blob]",
+                        "ContainerName":"[variables('containerName')]",
+                        "KeyVaultUrl": "[reference(concat('Microsoft.KeyVault/vaults/',parameters('keyVaultName')),'2018-02-14').vaultUri]",                        
+                        "AzureAd:CallbackPath":"/signin-oidc",
+                        "AzureAd:Instance":"[parameters('aadInstance')]",
+                        "AzureAd:Domain":"[parameters('aadDomain')]",
+                        "AzureAd:TenantId":"[parameters('aadTenantId')]",
+                        "AzureAd:ClientId":"[parameters('aadClientId')]"
+                    }
+                },
+                {
+                    "apiVersion": "2018-02-01",
+                    "name": "web",
+                    "type": "sourcecontrols",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/Sites', parameters('webAppName'))]",
+                        "[resourceId('Microsoft.Web/Sites/config', parameters('webAppName'),'appsettings')]"
+                    ],
+                    "properties": {
+                        "RepoUrl": "https://github.com/kaevans/aspnet-mvc-keyvault-storage.git",

```

Comment:
you cannot deploy from source of a personal repo in these samples... you can add the code here, use webdeploy (and add here) or just remove this resource